ios/RNSScreenStackHeaderConfigComponentView.mm

```diff
@@ -0,0 +1,679 @@
+      CGPoint convertedPoint = [_screenView.reactSuperview convertPoint:point toView:headerComponent];
+
+      UIView *hitTestResult = [headerComponent hitTest:convertedPoint withEvent:event];
+      if (hitTestResult != nil) {
+        return hitTestResult;
+      }
+    }
+  }
+  return nil;
+}
+
+- (void)updateViewControllerIfNeeded
+{
+  UIViewController *vc = _screenView.controller;
+  UINavigationController *nav = (UINavigationController *)vc.parentViewController;
+  UIViewController *nextVC = nav.visibleViewController;
+  if (nav.transitionCoordinator != nil) {
+    // if navigator is performing transition instead of allowing to update of `visibleConttroller`
+    // we look at `topController`. This is because during transitiong the `visibleController` won't
+    // point to the controller that is going to be revealed after transition. This check fixes the
+    // problem when config gets updated while the transition is ongoing.
+    nextVC = nav.topViewController;
+  }
+
+  if (vc != nil && (nextVC == vc)) {
+    [RNSScreenStackHeaderConfigComponentView updateViewController:self.screenView.controller
+                                                       withConfig:self
+                                                         animated:YES];
+  }
+}
+
+- (void)layoutNavigationControllerView
+{
+  UIViewController *vc = _screenView.controller;
+  UINavigationController *navctr = vc.navigationController;
+  [navctr.view setNeedsLayout];
+}
+
++ (void)setAnimatedConfig:(UIViewController *)vc withConfig:(RNSScreenStackHeaderConfigComponentView *)config
+{
+  UINavigationBar *navbar = ((UINavigationController *)vc.parentViewController).navigationBar;
+  // It is workaround for loading custom back icon when transitioning from a screen without header to the screen which
+  // has one. This action fails when navigating to the screen with header for the second time and loads default back
+  // button. It looks like changing the tint color of navbar triggers an update of the items belonging to it and it
+  // seems to load the custom back image so we change the tint color's alpha by a very small amount and then set it to
+  // the one it should have.
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_14_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_14_0
+  // it brakes the behavior of `headerRight` in iOS 14, where the bug desribed above seems to be fixed, so we do nothing
+  // in iOS 14
+  if (@available(iOS 14.0, *)) {
+  } else
+#endif
+  {
+    [navbar setTintColor:[config.color colorWithAlphaComponent:CGColorGetAlpha(config.color.CGColor) - 0.01]];
+  }
+  [navbar setTintColor:config.color];
+
+#if defined(__IPHONE_OS_VERSION_MAX_ALLOWED) && defined(__IPHONE_13_0) && \
+    __IPHONE_OS_VERSION_MAX_ALLOWED >= __IPHONE_13_0
+  if (@available(iOS 13.0, *)) {
+    // font customized on the navigation item level, so nothing to do here
+  } else
+#endif
+  {
```

Comment:
I have no idea why it works. Parentheses look completely off 🤔 . Same a few lines above.